### DISCUSSION TITLE
Positioning in BoxLayout

### DISCUSSION THREAD

**Discussion by RafaRioFal4699** [kivy/kivy-github-discussions#12] [Q&A] [answered]
I would like to know how to position elements inside a BoxLayout section, specifically an Image.

BoxLayout documentation says:

> Position hints are partially working, depending on the orientation:
>  ·  If the orientation is vertical: x, right and center_x will be used.
>  · If the orientation is horizontal: y, top and center_y will be used.

This makes me think there's no way to horizontally position an element inside a horizontal BoxLayout. Is that true? What is the best way to proceed?

In addition, Image documentation directly says an Image is automatically center-aligned. That means you cannot position yourself an image?

Sorry for the dummy questions, the documentation is a bit confusing 😅

**Comment by Julian-O** [comment]
A horizontal BoxLayout is a layout you use when you've decided you want the items laid directly next to each other, left to right. It will put the second widget right next to the first one.

So, when you say "there's no way to horizontally position an element inside a horizontal BoxLayout", that's kind of right (the hints are ignored) and kind of wrong (you precisely horizontally positioned all the elements when you chose a horizontal BoxLayout). You may be looking for a different layout to meet your needs.

**Reply by RafaRioFal4699** [reply]
So you mean if the size of the widget is less than the space dedicated to that section of BoxLayout, I cannot position freely in that space unless I put the widget inside another layout?

**Reply by Julian-O** [reply, marked as answer]
As I understand (and we are getting to my limits): The BoxLayout will, if it can, fill itself with subwidgets. (It may have no choice if all the subwidgets specify a size.)

But putting the widget inside another Layout to control how it fits in the space allocated by the Box Layout is a sensible approach.

**Reply by RafaRioFal4699** [reply]
Okay thanks, I got it!. I'll close the discussion.

**Comment by Julian-O** [comment]
The actual image is centred within the Image widget's bounding box, but you can position the image widget wherever you want (e.g. with a layout) so you still have control.

Imagine a poster stuck to a poster board. The poster is always centred inside the posterboard, but the posterboard can be the same size as the picture and put anywhere on the wall.
